From f2b4beac41e109b5129325def83f16d676b4f60c Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 25 Sep 2020 10:40:18 +0800
Subject: [PATCH] ARM: rockchip_defconfig: merge r/android-4.19 android base
 config

https://android.googlesource.com/kernel/configs
commit 5f72fe526dfe ("Merge "Require CONFIG_ANDROID_BINDERFS for R launching devices." into rvc-dev")

android-base.config and android-base-conditional.xml:

+CONFIG_IKHEADERS=m
-CONFIG_RT_GROUP_SCHED=y
+CONFIG_NAMESPACES=y
-CONFIG_MODULE_FORCE_UNLOAD=y
+CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG=y
-CONFIG_IP6_NF_NAT=y
-CONFIG_IP6_NF_TARGET_MASQUERADE=y
+CONFIG_BPF_JIT=y
+# CONFIG_FW_CACHE is not set
+CONFIG_DM_SNAPSHOT=y
+CONFIG_DUMMY=y
+CONFIG_VETH=y
+# CONFIG_DEVMEM is not set
+CONFIG_EXT4_FS_POSIX_ACL=y
-# CONFIG_F2FS_FS_POSIX_ACL is not set
+CONFIG_FS_VERITY=y
+CONFIG_FS_VERITY_BUILTIN_SIGNATURES=y
+CONFIG_STATIC_USERMODEHELPER=y
+CONFIG_STATIC_USERMODEHELPER_PATH=""

Except CONFIG_ANDROID_BINDERFS=y which do not supported by Android before 11.
CONFIG_ANDROID_BINDERFS is set by Android.

Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: I8de7e0392c1fb0fe05f957f1f2710b295cafea07
---
 arch/arm/configs/rockchip_defconfig | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/arch/arm/configs/rockchip_defconfig b/arch/arm/configs/rockchip_defconfig
index 05dfc5e77d97..f6f22bdbdad9 100644
--- a/arch/arm/configs/rockchip_defconfig
+++ b/arch/arm/configs/rockchip_defconfig
@@ -12,17 +12,18 @@ CONFIG_TASK_IO_ACCOUNTING=y
 CONFIG_PSI=y
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
+CONFIG_IKHEADERS=m
 CONFIG_LOG_BUF_SHIFT=19
 CONFIG_CGROUPS=y
 CONFIG_MEMCG=y
 CONFIG_MEMCG_SWAP=y
 CONFIG_BLK_CGROUP=y
 CONFIG_CGROUP_SCHED=y
-CONFIG_RT_GROUP_SCHED=y
 CONFIG_CGROUP_FREEZER=y
 CONFIG_CPUSETS=y
 CONFIG_CGROUP_CPUACCT=y
 CONFIG_CGROUP_BPF=y
+CONFIG_NAMESPACES=y
 CONFIG_BLK_DEV_INITRD=y
 # CONFIG_RD_BZIP2 is not set
 # CONFIG_RD_LZMA is not set
@@ -81,7 +82,6 @@ CONFIG_CRYPTO_AES_ARM_BS=y
 CONFIG_JUMP_LABEL=y
 CONFIG_MODULES=y
 CONFIG_MODULE_UNLOAD=y
-CONFIG_MODULE_FORCE_UNLOAD=y
 CONFIG_MODVERSIONS=y
 CONFIG_PARTITION_ADVANCED=y
 CONFIG_CFQ_GROUP_IOSCHED=y
@@ -168,6 +168,7 @@ CONFIG_NETFILTER_XT_MATCH_POLICY=y
 CONFIG_NETFILTER_XT_MATCH_PKTTYPE=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA2=y
+CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG=y
 CONFIG_NETFILTER_XT_MATCH_SOCKET=y
 CONFIG_NETFILTER_XT_MATCH_STATE=y
 CONFIG_NETFILTER_XT_MATCH_STATISTIC=y
@@ -197,8 +198,6 @@ CONFIG_IP6_NF_FILTER=y
 CONFIG_IP6_NF_TARGET_REJECT=y
 CONFIG_IP6_NF_MANGLE=y
 CONFIG_IP6_NF_RAW=y
-CONFIG_IP6_NF_NAT=y
-CONFIG_IP6_NF_TARGET_MASQUERADE=y
 CONFIG_BRIDGE_NF_EBTABLES=y
 CONFIG_BRIDGE_EBT_BROUTE=y
 CONFIG_L2TP=y
@@ -219,6 +218,7 @@ CONFIG_NET_EMATCH_U32=y
 CONFIG_NET_EMATCH_META=y
 CONFIG_NET_EMATCH_TEXT=y
 CONFIG_NET_CLS_ACT=y
+CONFIG_BPF_JIT=y
 CONFIG_BT=y
 CONFIG_BT_RFCOMM=y
 CONFIG_BT_RFCOMM_TTY=y
@@ -231,6 +231,7 @@ CONFIG_BT_HCIUART_H4=y
 CONFIG_RFKILL=y
 CONFIG_DEVTMPFS=y
 CONFIG_FW_LOADER_USER_HELPER=y
+# CONFIG_FW_CACHE is not set
 CONFIG_DMA_CMA=y
 CONFIG_ZRAM=y
 CONFIG_BLK_DEV_LOOP=y
@@ -248,13 +249,16 @@ CONFIG_SCSI_SCAN_ASYNC=y
 CONFIG_MD=y
 CONFIG_BLK_DEV_DM=y
 CONFIG_DM_CRYPT=y
+CONFIG_DM_SNAPSHOT=y
 CONFIG_DM_UEVENT=y
 CONFIG_DM_VERITY=y
 CONFIG_DM_VERITY_AVB=y
 CONFIG_DM_VERITY_FEC=y
 CONFIG_DM_ANDROID_VERITY_AT_MOST_ONCE_DEFAULT_ENABLED=y
 CONFIG_NETDEVICES=y
+CONFIG_DUMMY=y
 CONFIG_TUN=y
+CONFIG_VETH=y
 # CONFIG_NET_VENDOR_ALACRITECH is not set
 # CONFIG_NET_VENDOR_AMAZON is not set
 # CONFIG_NET_VENDOR_AQUANTIA is not set
@@ -402,6 +406,7 @@ CONFIG_INPUT_RK805_PWRKEY=y
 # CONFIG_SERIO is not set
 # CONFIG_VT is not set
 # CONFIG_LEGACY_PTYS is not set
+# CONFIG_DEVMEM is not set
 CONFIG_SERIAL_8250=y
 # CONFIG_SERIAL_8250_DEPRECATED_OPTIONS is not set
 CONFIG_SERIAL_8250_CONSOLE=y
@@ -618,11 +623,13 @@ CONFIG_OPTEE=y
 CONFIG_RK_NAND=y
 CONFIG_RK_HEADSET=y
 CONFIG_EXT4_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
 CONFIG_EXT4_FS_SECURITY=y
 CONFIG_F2FS_FS=y
-# CONFIG_F2FS_FS_POSIX_ACL is not set
 CONFIG_F2FS_FS_SECURITY=y
 CONFIG_F2FS_FS_ENCRYPTION=y
+CONFIG_FS_VERITY=y
+CONFIG_FS_VERITY_BUILTIN_SIGNATURES=y
 CONFIG_QUOTA=y
 CONFIG_QUOTA_NETLINK_INTERFACE=y
 CONFIG_QFMT_V2=y
@@ -651,6 +658,8 @@ CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_LSM_MMAP_MIN_ADDR=4096
 CONFIG_HARDENED_USERCOPY=y
+CONFIG_STATIC_USERMODEHELPER=y
+CONFIG_STATIC_USERMODEHELPER_PATH=""
 CONFIG_SECURITY_SELINUX=y
 CONFIG_TEE_SUPPORT=y
 CONFIG_CRYPTO_TWOFISH=y
-- 
2.35.3

